### PR TITLE
village-tracker docs !

### PR DESCRIPTION
**Pull request by mixmix**
this improves : 
- `README.md`
  - note there are still undocumented things but I've put placeholders
- `PLUGINS.md`
  - this is looking real good
- some readability of code for beginners
  - moved a couple things into `util.js`
  - made `index.js` look a little simpler for someone new

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -38,16 +24,16 @@ function find(ary, test) {
...
-module.exports = function (plugins, defaults) {
-
+module.exports = function (plugins, defaultConfig) {
   function create (opts) {
-    opts = merge(merge({}, defaults), opts)
+    opts = merge(merge({}, defaultConfig), opts)
     //change event emitter to something with more rigorous security?
     var api = new EventEmitter()
     create.plugins.forEach(function (plug) {
+      // mix: uhhh where is create.createClient defined?
```
Comment (mixmix) [inline api.js]: hey @dominictarr what is this and where does it come from... I searched files and couldn't find it anywhere

Comment (dominictarr) [reply api.js]: oh, looks like that code has been removed. now you use .connect instead (provided by core.js)

File: README.md [hunk 2]
```diff
@@ -122,6 +92,34 @@ node.auth.hook(function (auth, args) {
...
 
+### `app.close()`
+
+close the app!
+
+Optionally takes `(err, callback)`
+
+----
+
+## TODO document
```
Comment (dominictarr) [inline README.md]: it's good for a reader to know that there are things that are not documented yet, but in the interest of actually getting these documented, it's better if they are in an issue

**Comment by dominictarr** [comment]
merged into 6.1.1

**Pull request merged**